rpa-builder/6.1/modules/ROOT/pages/toolbox-mail-operations-mail-session-outlook-with-oauth.adoc

```diff
@@ -0,0 +1,196 @@
+* *Use Different Credentials*
++
+(Boolean) Select this option to set
+different credentials to those specified under Incoming Mail - User
+account for outgoing mail.
+
+== Outbound Variables
+
+The _Mail Session_ Action Step has no outbound variables.
+
+== Wizard
+
+image:mail-session-outlook-oauth-wizard.png[The Mail Session (Outlook with OAuth) Action Step Wizard, 60%, 60%]
+
+The wizard contains the following sections: *Operation Mode*, *OAuth Authentication*, *Incoming Mail*, and *Outgoing Mail*.
+
+=== Operation Mode
+
+In *Operation Mode* you determine what happens and which area activates in the Mail Session. The
+following modes are available:
+
+* *Read and send mails*
++
+Enables you to configure the Incoming and Outgoing
+Mail areas so you can read and send mails.
+* *Read mails from inbox*
++
+Enables editing for the Incoming Mail section, so that you can only read mails.
+* *Send mails via SMTP*
++
+Enables editing for the Outgoing Mail section , so that you can only send mails.
++
+Note that if you selected the _Send mails via SMTP_ mode, you can't use the Read mail Action Step in the Mail Session.
+This is also true if you have selected the _Read mails from inbox_ mode. In this case, you can't use the Send Mail Action Step.
+
+=== OAuth Authentication
+
+The *OAuth Authentication* section enables you to create a set of tokens, that don't rely on a user’s password to authenticate with the email server. After you create the refresh token, you can use it to repeatedly log in to the services without asking the user's password again.
+
+Follow these steps to authenticate with your credentials:
+
+. Complete the *OAuth Authentication* configuration by specifying the required properties.
++
+Refer to <<properties, OAuth Authentication Settings>> for a description of each property.
+. Click *Authenticate*.
++
+This step starts the authentication process with the OAuth Host, which opens the login page in your default browser.
+. In the OAuth Host’s login page, complete the login process.
+. Close the browser.
+
+After completing the last step, the wizard shows that you are authenticated:
+image:email-outlook-authentication.png[The Logged in confirmation message in the authentication settings window, 50%, 50%]
+
+==== Generating a OAuth Refresh Token
+
+Instead of authenticating via the *Mail Session (Outlook with OAuth) wizard*, you can pin the refresh token if you obtain it externally. In this case, you still need to fill (or pin) the rest of the fields.
+
+To generate a and configure a refresh token:
+
+. In the *Tools* menu, select *Generate OAuth Refresh Token*.
++
+image:generate-oauth-token.png[The Generate OAuth Refresh Token options in the Tools menu, 40%, 40%]
+. Complete the *OAuth Token Creator* form that appears.
++
+image:generate-token-form.png[The OAuth Token Creator Form, 60%, 60%]
++
+If you already configured a Mail Session (Outlook with OAuth), the data automatically fills with the last data you entered in the wizard.
```

Comment:
This doesn't happen automatically, the user has to press the button "Save data to generate Tokens via the Tools menu" within the Outlook with OAuth Wizard before. Only then will the data in this dialog be filled.